### PR TITLE
Fix memory leaks SiteCreation Segments and Domains screens

### PR DESCRIPTION
**Pull request by malinajirka**
I've completely forgotten about these changes @zwarm and I implemented during a pair programming session a couple of weeks ago.

Fixes leaks in Segments and Domains Site Creation's screens. (PreviewFragment is tbd)

Fragment can live even after its views were destroyed (eg on backstack). If the fragment keeps references to views after `onViewDestroyed` is invoked, all the views are leaked. This PR replaces class fields with kotlin synthetic accessors.
One extra change is, that the setupContent method is invoked from within "onViewCreated". It's the correct place anyway, since "onCreateView" should be taking care of view creation but not initialisation.

To test:
1. My Site
2. Switch Site
3. Turn on airplane mode
4. Click on "+" and "Create WordPress.com site"
5. Observe the error screen is shown
6. Click on retry and observe a progress is shown for a bit and error screen is shown again
7. Turn off airplane mode and click on retry
8. Segments screen is shown
9. Select a segment
10. Turn on airplane mode and type something
11. Error is shown
12. Turn off airplane mode
13. Click on retry
14. Observe results are shown
15. Select a domain and click on Create Site button
PR submission checklist:

- [x] I have considered adding unit tests where possible.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/ui/sitecreation/domains/SiteCreationDomainsFragment.kt [hunk 1]
```diff
@@ -88,20 +77,17 @@ class SiteCreationDomainsFragment : SiteCreationBaseFormFragment() {
         super.onViewStateRestored(savedInstanceState)
         // we need to set the `onTextChanged` after the viewState has been restored otherwise the viewModel.updateQuery
         // is called when the system sets the restored value to the EditText which results in an unnecessary request
-        searchInputWithHeader.onTextChanged = { viewModel.updateQuery(it) }
+        searchInputWithHeader?.onTextChanged = { viewModel.updateQuery(it) }
     }
 
     private fun initRecyclerView(rootView: ViewGroup) {
```
Comment (oguzkocer) [inline WordPress/src/main/java/org/wordpress/android/ui/sitecreation/domains/SiteCreationDomainsFragment.kt:83]: It looks like we don't need this parameter anymore.

```suggestion
    private fun initRecyclerView() {
```

File: WordPress/src/main/java/org/wordpress/android/ui/sitecreation/SiteCreationBaseFormFragment.java [hunk 2]
```diff
@@ -47,14 +47,15 @@ protected ViewGroup createMainView(LayoutInflater inflater, ViewGroup container,
...
         ViewGroup rootView = createMainView(inflater, container, savedInstanceState);
-        setupContent(rootView);
         return rootView;
     }
 
     @Override
     public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
         super.onViewCreated(view, savedInstanceState);
 
+        setupContent((ViewGroup) getView().getRootView());
```
Comment (oguzkocer) [inline WordPress/src/main/java/org/wordpress/android/ui/sitecreation/SiteCreationBaseFormFragment.java:57]: Would it be better to use the `view` argument here instead? AS is complaining about a potential NPE here.

```suggestion
        setupContent((ViewGroup) view);
```

File: WordPress/src/main/java/org/wordpress/android/ui/sitecreation/SiteCreationBaseFormFragment.java [hunk 3]
```diff
@@ -47,14 +47,15 @@ protected ViewGroup createMainView(LayoutInflater inflater, ViewGroup container,
     @Override
     public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
         ViewGroup rootView = createMainView(inflater, container, savedInstanceState);
-        setupContent(rootView);
         return rootView;
```
Comment (oguzkocer) [inline WordPress/src/main/java/org/wordpress/android/ui/sitecreation/SiteCreationBaseFormFragment.java:50]: Nitpick: AS is complaining about the unnecessary `rootView` variable. 😿

File: WordPress/src/main/java/org/wordpress/android/ui/sitecreation/SiteCreationBaseFormFragment.java [hunk 4]
```diff
@@ -47,14 +47,15 @@ protected ViewGroup createMainView(LayoutInflater inflater, ViewGroup container,
...
     }
 
     @Override
     public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
         super.onViewCreated(view, savedInstanceState);
 
+        setupContent((ViewGroup) getView().getRootView());
+
         Toolbar toolbar = view.findViewById(R.id.toolbar_main);
         ((AppCompatActivity) getActivity()).setSupportActionBar(toolbar);
```
Comment (oguzkocer) [inline WordPress/src/main/java/org/wordpress/android/ui/sitecreation/SiteCreationBaseFormFragment.java:60]: This is unrelated to this PR, but AS is complaining about a potential NPE here, should we change it to:
```suggestion
        ((AppCompatActivity) requireActivity()).setSupportActionBar(toolbar);
```

File: WordPress/src/main/java/org/wordpress/android/ui/sitecreation/SiteCreationBaseFormFragment.java (hunk 2 above)
Comment (malinajirka) [reply WordPress/src/main/java/org/wordpress/android/ui/sitecreation/SiteCreationBaseFormFragment.java:56]: I think passing `view` would also probably work, but I'd rather not change this just to be safe. (+ to save some time testing the change)

**Comment by malinajirka** [comment]
Thanks for the review @oguzkocer! I've pushed the proposed changes with exception of the `view.getRootView -> view` replacement. I'm not sure if you want to review it again. Let me know, thanks!

**Comment by planarvoid** [comment]
Changes in 41cabe2 were already reviewed in another PR so feel free to ignore them in this review

File: WordPress/src/main/java/org/wordpress/android/ui/sitecreation/SiteCreationBaseFormFragment.java (hunk 2 above)
Comment (oguzkocer) [reply WordPress/src/main/java/org/wordpress/android/ui/sitecreation/SiteCreationBaseFormFragment.java:56]: I actually tested this before I made the suggestion, but it was not an important point. Thank you for updating the rest of the code and making AS happy.

**Pull request merged**